src/models/HybridSystem.jl

```diff
@@ -0,0 +1,172 @@
+"""
+Representation of a Hybrid System that collects renewable generation, load, thermal generation
+and storage
+"""
+mutable struct HybridSystem{
+    T <: ThermalGen,
+    L <: ElectricLoad,
+    S <: Storage,
+    R <: RenewableGen,
+} <: StaticInjection
+    name::String
+    available::Bool
+    status::Bool
+    bus::Bus
+    active_power::Float64
+    reactive_power::Float64
+    thermal_unit::T
+    electric_load::L
+    storage::S
+    renewable_unit::R
+    # PCC Data
+    "Thermal limited MVA Power Output of the unit. <= Capacity"
+    pcc_impedance::ComplexF64
+    rating::Float64
+    input_active_power_limits::NamedTuple{(:min, :max), Tuple{Float64, Float64}}
+    output_active_power_limits::NamedTuple{(:min, :max), Tuple{Float64, Float64}}
+    reactive_power_limits::Union{Nothing, Min_Max}
+    base_power::Float64
```

Comment:
How will this relate to the `base_power` of the components in the `thermal_unit` and other fields? Will this `base_power` only apply to the local fields while the sub-components will have their own?

Reply:
This would be the base power of the PCC interconnection either given by the transformer or the output inverter. On the practical side, it is needed to scale the PCC input/output limits and rating.